File: tests/files/extract_var_works_after.rb
def foo()
    foo = "foo"
    print foo

    while true do
        print foo
        break
    end

    if true then
        print foo
    else
        print foo
    end

    for i in range(0, 5) do
        print foo
    end

    [:a, :b].each do |item|
        print foo
    end

    [:a, :b].each {|item|
        print foo
    }
end

File: tests/files/extract_var_works_after.js
const foo = "foo";
function bar() {
  console.log(foo);
}

console.log(foo);

while (false) {
  console.log(foo);
}

do {
  console.log(foo);
} while (false);

if (true) {
  console.log(foo);
} else {
  console.log(foo);
}

for (let i = 0; i < 5; i++) {
  console.log(foo);
}
const baz = () => {
  console.log(foo);
};
